src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -11,37 +11,40 @@ import {
   AssetCalibrationForecastKey,
   AssetCalibrationQuery,
   AssetCalibrationTimeBasedGroupByType,
-  AssetModel,
-  AssetsResponse,
   CalibrationForecastResponse,
 } from './types';
-import { SystemMetadata } from "../system/types";
-import { defaultOrderBy, defaultProjection } from "../system/constants";
+import { AssetModel, AssetsResponse } from 'core/types';
 
 export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQuery> {
+  public defaultQuery = {
+    groupBy: [],
+    filter: ''
+  };
+
   constructor(
     readonly instanceSettings: DataSourceInstanceSettings,
     readonly backendSrv: BackendSrv = getBackendSrv(),
-    readonly templateSrv: TemplateSrv = getTemplateSrv()
+    readonly templateSrv: TemplateSrv = getTemplateSrv(),
   ) {
     super(instanceSettings, backendSrv, templateSrv);
   }
 
-  defaultQuery = {
-    groupBy: [],
-  };
-
   baseUrl = this.instanceSettings.url + '/niapm/v1';
 
   async runQuery(query: AssetCalibrationQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+    if (this.templateSrv.containsTemplate(query.filter)) {
```

Comment:
The `containsTemplate check shouldn't be necessary—if there are no variables it will just return the same string

Reply:
Updated as suggested, only checking for the filter to actually have values.

Reply:
I'm not even sure if that check is necessary either. The `defaultQuery` should ensure that the filter is never null/undefined.